okio/src/appleMain/kotlin/okio/Buffer.kt

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (C) 2020 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okio
+
+import kotlinx.cinterop.ByteVarOf
+import kotlinx.cinterop.CPointer
+import kotlinx.cinterop.plus
+import kotlinx.cinterop.addressOf
+import kotlinx.cinterop.usePinned
+import platform.Foundation.NSData
+import platform.Foundation.NSMutableData
+import platform.Foundation.create
+import platform.darwin.NSInteger
+import platform.posix.memcpy
+
+fun Buffer.write(
```

Comment:
What does this look like from Obj-C or Swift?

Reply:
```swift
import Foundation
import Okio

let hello = "Hello".data(using: .utf8)!

let buffer = Buffer()
buffer.write(source: hello, offset: 0, byteCount_: Int32(hello.count))

let data = buffer.readNSData(byteCount: UInt64(hello.count))
print(String.init(data: data, encoding: .utf8)!)
```

Reply:
1. Weird that `byteCount` required different conversions for write and read, will look into changing signatures to not require any conversions.
2. Should probably drop `NS` to follow Swift conventions, maybe even switch to just `read()`.